src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -129,6 +136,44 @@ public function panelAction(Request $request, $token)
     }
 
     /**
+     * Replays an action.
+     *
+     * @param Request $request The current HTTP Request
+     * @param string  $token   The profiler token
+     *
+     * @return RedirectResponse A RedirectResponse instance
+     */
+    public function replayAction(Request $request, $token)
+    {
+        $profile = $this->profiler->loadProfile($token);
+
+        $replayer = new Client($kernel = clone $this->kernel);
```

Comment:
you don't need a client to perform the Request. You could use `$kernel->handle()`